llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -813,16 +813,102 @@ static bool hasComputableBounds(PredicatedScalarEvolution &PSE, Value *Ptr,
   return AR->isAffine();
 }
 
+/// Try to compute the stride for \p AR. Used by getPtrStride.
+static std::optional<int64_t>
```

Comment:
Can you recommit this extraction and rebase?

Reply:
Done in 424fcc5df7d7ba1db3d83023885dcbaf33a49b39, thanks